StatsService.js

```diff
@@ -7,12 +7,118 @@ import axios from 'axios';
 
 const headers = {Accept: 'application/ld+json, application/json'};
 
-export class StatsService {
-  constructor() {
-  }
+/**
+ * A subscription is an Object with an id and an updater function that is used
+ * to make a chart. A ChartController implements both id and updater.
+ * @typedef {Object} Subscription
+ * @property {String} id - a unique id for the subscription
+ * @property {Function} updater - a function called on update
+ */
 
+class StatsService {
+  /**
+   * Stats Service is a singleton.
+   * It polls the stats api using an interval.
+   * It is stored in window.bedrock.statsService.
+   *
+   * @param {Object} options - Options for the StatsService.
+   * @param {number} [options.poll = 1000] -
+   * time in ms for each request to the api.
+   *
+   * @returns {StatsService} Either an existing one or a new one.
+   */
+  constructor({poll = 1000} = {}) {
+    if(!window.bedrock) {
```

Comment:
Why are we using globals here? Is this the only path forward that you found for completing this section of code?

Reply:
what would you suggest here? How for instance did angular handle singletons? there has to be a place to look for the singleton so I used window. more than happy to revise.